src/server/db_slice.cc

```diff
@@ -1158,6 +1143,11 @@ DbSlice::PrimeItAndExp DbSlice::ExpireIfNeeded(const Context& cntx, PrimeIterato
   const_cast<DbSlice*>(this)->PerformDeletion(Iterator(it, StringOrView::FromView(key)),
                                               ExpIterator(expire_it, StringOrView::FromView(key)),
                                               db.get());
+  // Replicate expiry
```

Comment:
what is the reason you moved this here?

Reply:
Good question. The answer is `preemptions` because `heartbeat` is not `transactional`. If we first write to the journal, preempt and then delete, we risk another transaction coming in between the preemption and before the deletion. This is not good.

Reply:
RecordExpiry can not preempt, the await is set to false
Does this assumption breaks now?